ethash/src/compute.rs

```diff
@@ -85,6 +86,7 @@ pub type H256 = [u8; 32];
 pub struct Light {
 	block_number: u64,
 	cache: Vec<Node>,
+	eth_compute: Mutex<EthCompute>,
```

Comment:
This needs to be a threadsafe, due to it's use by Ethash - because `Engine` has a `Sync` constraint.